eng/pipelines/partner-release.yml

```diff
@@ -36,11 +50,12 @@ steps:
 - publish: $(Artifacts)
   artifact: packages-signed
 
-- template: tools/gpg/gpg.yml@azure-sdk-build-tools
-- template: /eng/pipelines/templates/steps/java-publishing.yml
-  parameters:
-    ArtifactDirectory: $(Artifacts)
-    Target: MavenCentral
-    BuildToolsPath: $(Pipeline.Workspace)/azure-sdk-build-tools
-    JavaRepoRoot: $(Pipeline.Workspace)/azure-sdk-for-java
-    StageOnly: $(StageOnly)
+- ${{ if eq(parameters.ShouldPublish, 'true') }}:
+  - template: tools/gpg/gpg.yml@azure-sdk-build-tools
```

Comment:
In a follow-up we should try to do the gpg signing all the time and upload those as artifacts so we have the exact files we plan to upload.